src/EditorFeatures/Core/InlineRename/CommandHandlers/AbstractRenameCommandHandler_RenameHandler.cs

```diff
@@ -92,14 +105,38 @@ private void ExecuteRenameWorker(RenameCommandArgs args, CommandExecutionContext
             // There can be zero selectedSpans in projection scenarios.
             if (selectedSpans.Count != 1)
             {
-                ShowErrorDialog(workspace, EditorFeaturesResources.You_must_rename_an_identifier);
+                await ShowErrorDialogAsync(workspace, EditorFeaturesResources.You_must_rename_an_identifier, _threadingContext).ConfigureAwait(false);
                 return;
             }
 
-            var sessionInfo = _renameService.StartInlineSession(document, selectedSpans.Single().Span.ToTextSpan(), cancellationToken);
+            var sessionInfo = await _renameService.StartInlineSessionAsync(document, selectedSpans.Single().Span.ToTextSpan(), cancellationToken).ConfigureAwait(false);
             if (!sessionInfo.CanRename)
             {
-                ShowErrorDialog(workspace, sessionInfo.LocalizedErrorMessage);
+                await ShowErrorDialogAsync(workspace, sessionInfo.LocalizedErrorMessage, _threadingContext).ConfigureAwait(false);
```

Comment:
passing an instance field to an instance method seems unnecessary (_threadinContext).